### PR TITLE
Add Data Explorer max number of items listed

### PR DESCRIPTION
**Pull request by jason-seqera**
- https://seqera.atlassian.net/browse/EDU-161

Targets 24.1+

### CODE REVIEW

**Comment by jason-seqera** [comment]
Hi @endre-seqera, I've added these Data Explorer values (the MB limit was already present) to the docs.

- Do we know if there are any equivalent ENV variables for this? I can't tell
- What happens if someone specifies `25<space>MB` or `25` or `25K`? How is this string processed to arrive at a limit in bytes? Can someone use the typical suffixes, like MB, mb, KB, KiB, ect? Is it only MB/KB/GB?

Thanks!

File: platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+-
+  tower.yml:            '`tower.content.max-file-size`'
+  Description: >
+    You can set the Data Explorer downloadable file size limit. Increasing this value may degrade Platform performance.
+  Value:                'Default: `25MB`'
+-
+  tower.yml:            '`tower.data.data-link-items.tree-list.max-allowed`'
+  Description: >
+    You can set the max number of items listed in Data Explorer.
```
Comment (justinegeffen) [inline platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml:10]:
```suggestion
    You can set the maximum number of items listed in Data Explorer.
```

File: platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+---
+-
+  tower.yml:            '`tower.content.max-file-size`'
+  Description: >
+    You can set the Data Explorer downloadable file size limit. Increasing this value may degrade Platform performance.
```
Comment (justinegeffen) [inline platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml:5]: Curious as to whether this has a maximum limit?

Comment (endre-seqera) [reply platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml:5]: nope, there is no enforcing of a maximum limit
so basically customers are allowed to set a really high value and experience super slow downloads for these big files

File: platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  Value:                'Default: `25MB`'
+-
+  tower.yml:            '`tower.data.data-link-items.tree-list.max-allowed`'
+  Description: >
+    You can set the maximum number of items listed in Data Explorer.
```
Comment (endre-seqera) [inline platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml:10]: minor: I liked the above phrasing and would add here as well: `Increasing this value may degrade Platform performance.`

**Comment by endre-seqera** [comment]
Hi @jason-seqera, these are great questions! 
As I just had approximate knowledge of these, I needed to look into how this exactly works and learned some new things in the process :) 

>     * Do we know if there are any equivalent ENV variables for this? I can't tell

Yes, there are. Micronaut allows overriding any property via environment variables. There are some specific rules on how environment variables are binded/mapped to properties which can be seen in the [docs of micronaut](https://docs.micronaut.io/latest/guide/index.html#propertySource) - section on `Property Value Binding` - but the short summary is that underscore `_` gets converted to all dot and hypen combinations. So, we could use env variables like:  TOWER_CONTENT_MAX_FILE_SIZE.

But maybe it's not worth to explain all this in docs, as setting this property in the application config file (.yml file) is more than enough?


>     * What happens if someone specifies `25<space>MB` or `25` or `25K`? How is this string processed to arrive at a limit in bytes? Can someone use the typical suffixes, like MB, mb, KB, KiB, ect? Is it only MB/KB/GB?

This value is annotated with `@ReadableBytes`, and this annotation determines how the value is actually converted. This is described in the micronaut docs as well (see Readable Bytes section). The answer to your specific questions are:
- space is not allowed between number and byte shorthand syntax (25<space>MB will result in a conversion error - unable to convert 25<space> to long)
- the only allowed suffixes are: `KB, MB, GB`, but the value is automatically converted to uppercase so also these variations are allower `kb, kB, Kb, mb, mB, Mb, gb, Gb, gB`
- if a byte shorthand syntax suffix is not present - example `25` - it is assumed that the value is in bytes


> Is this available in all versions of Platform that include Data Explorer?

- `tower.content.max-file-size` was introduced starting from `v22.1_1d4defa` and `v23.3.0-BETA1-enterprise`
- `tower.data.data-link-items.tree-list.max-allowed` was introduced starting from `v24.1.0-cycle17_ed131c8` and `v24.1.0-RC1-enterprise`

File: platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml (hunk 2 above)
Comment (endre-seqera) [inline platform_versioned_docs/version-24.1/enterprise/configuration/configtables/data_features_yml.yml:5]: This value also controls the max file size limit for downloadable workflow reports (which is outside of DataExplorer)

**Comment by jason-seqera** [comment]
@endre-seqera, thanks!

**Comment by jason-seqera** [comment]
@endre-seqera @justinegeffen - I've made the suggested updates, thanks!

**Review by endre-seqera — APPROVED** [review]

**Pull request merged**